### PR TITLE
T5173 azure api keys

### PR DESCRIPTION
**Pull request by thinkst-pieter**
Add support for Azure tokens to canarytokens.org

### CODE REVIEW

File: queries.py [hunk 1]
```diff
@@ -210,6 +210,43 @@ def get_aws_keys(token=None, server=None):
...
+        url = '{token_url}?code={token_auth}'.format(token_url=token_url, token_auth=token_auth)
+        data = {
+            'token': token,
+            'domain': server
+        }
+
+        resp = requests.post(url=url, json=data)
+        if not resp:
+            log.error('Bad response from getting AZURE ID')
+            return False
```
Comment (jayjb) [inline queries.py:238]: it may be better to just
resp.raise_for_status()

File: httpd_site.py [hunk 2]
```diff
@@ -249,6 +250,31 @@ def render_POST(self, request):
...
+                    raise Exception()
+                keys = get_azure_id(token=canarytoken.value(), server=get_all_canary_domains()[0])
+                if not keys:
+                    response['Error'] = 4
+                    response['Error_Message'] = 'Failed to retrieve Azure ID. Please contact support@thinkst.com.'
+                    raise Exception()
+                azure_id_cert_file_name = request.args['azure_id_cert_file_name'][0]
+                if not azure_id_cert_file_name:
+                    raise Exception()
+                response['app_id'] = keys[0]
```
Comment (jayjb) [inline httpd_site.py:264]: you may be able to do 
```
response['app_id'] = canarydrop['app_id'] = keys[0]
```

File: queries.py (hunk 1 above)
Comment (thinkst-pieter) [reply queries.py:238]: May wan't to be careful as to what the azure server returns so as to not expose something. We have the logs on the azure side

File: httpd_site.py (hunk 2 above)
Comment (thinkst-pieter) [reply httpd_site.py:264]: Done

**Pull request merged**